src/main/perl/RuleBasedEditor.pm

```diff
@@ -0,0 +1,1175 @@
+            *$self->{LOG}->debug(2, "$function_name: array values sorted: ", join(",", @$attr_value));
+        }
+        $formatted_value = join " ", @$attr_value;
+
+    } elsif ($value_fmt == LINE_VALUE_HASH_KEYS) {
+        $formatted_value = join " ", sort keys %$attr_value;
+
+    } elsif (($value_fmt == LINE_VALUE_AS_IS) || ($value_fmt == LINE_VALUE_STRING_HASH)) {
+        $formatted_value = $attr_value;
+
+    } else {
+        *$self->{LOG}->error("$function_name: invalid value format ($value_fmt) (internal error)");
+    }
+
+    # Quote value if necessary
+    if (($line_fmt == LINE_FORMAT_SH_VAR) || ($line_fmt == LINE_FORMAT_ENV_VAR)) {
+        if (   (($formatted_value =~ /\s+/) && ($formatted_value !~ /^(["']).*\g1$/))
+            || ($value_fmt == LINE_VALUE_BOOLEAN)
+            || ($formatted_value eq ''))
+        {
+            *$self->{LOG}->debug(2, "$function_name: quoting value '$formatted_value'");
+            $formatted_value = '"' . $formatted_value . '"';
+        }
+    }
+
+    *$self->{LOG}->debug(2, "$function_name: formatted value >>>$formatted_value<<<");
+    return $formatted_value;
+}
+
+
+=pod
+
+=item _formatConfigLine
+
+This function formats a configuration line using keyword and value,
+according to the line format requested. Values containing spaces are
+quoted if the line format is not LINE_FORMAT_KEY_VAL.
+
+Arguments :
+    keyword : line keyword
+    value : keyword value (can be empty)
+    line_fmt : line format (see LINE_FORMAT_xxx constants)
+
+Return value:
+    A string corresponding to the line formatted according to line_fmt
+    or undef in case of an internal error (missing arguments)
+
+=cut
+
+sub _formatConfigLine
+{
+    my $function_name = "_formatConfigLine";
+    my ($self, $keyword, $value, $line_fmt) = @_;
+
+    unless ($keyword) {
+        *$self->{LOG}->error("$function_name: 'keyword' argument missing (internal error)");
+        return undef;
+    }
+    unless (defined($value)) {
+        *$self->{LOG}->error("$function_name: 'value' argument missing (internal error)");
+        return undef;
+    }
+    unless (defined($line_fmt)) {
+        *$self->{LOG}->error("$function_name: 'line_fmt' argument missing (internal error)");
+        return undef;
+    }
+
+    my $config_line = "";
+
+    if ($line_fmt == LINE_FORMAT_SH_VAR) {
+        $config_line = "$keyword=$value";
+    } elsif ($line_fmt == LINE_FORMAT_ENV_VAR) {
+        $config_line = "export $keyword=$value";
+    } elsif ($line_fmt == LINE_FORMAT_KEY_VAL_SETENV) {
+        $config_line = "setenv $keyword = $value";
+    } elsif ($line_fmt == LINE_FORMAT_KEY_VAL_SET) {
+        $config_line = "set $keyword = $value";
+    } elsif ($line_fmt == LINE_FORMAT_KEY_VAL) {
+        $config_line = $keyword;
+        $config_line .= " $value" if $value;
+        # In trust (shift.conf) format, there should be only one blank between
+        # tokens and no trailing spaces.
+        $config_line =~ s/\s\s+/ /g;
```

Comment:
I think it is useless. PRobably the intent was to avoid a replace when there was already one space (tabs should be unusual in the value).